Engines/Wine/Engine/Implementation/script.js

```diff
@@ -444,21 +447,25 @@ module.default = class WineEngine {
             ldPath = userData.ldPath + ldPath;
         }
 
+        let runtimePath = "";
+        let runtimePath64 = "";
+        if (this._useRuntime) {
+            runtimePath = this._wineEnginesDirectory + "/runtime/lib/";
+            runtimePath64 = this._wineEnginesDirectory + "/runtime/lib64/";
+        }
```

Comment:
Does it make sense to move this to another internal/private method? Something like:

```javascript
fetchRuntimePaths() {
   if (this._useRuntime) {
      return {
         runtimePath: `{this._wineEnginesDirectory}/runtime/lib/`,
         runtimePath64: `{this._wineEnginesDirectory}/runtime/lib64/`
      };
   } else {
      return { runtimePath: "", runtimePath64: "" };
   }
}
```

Afterwards we can fetch the runtime paths with:

```javascript
const { runtimePath, runtimePath64 } = this.fetchRuntimePaths();
```

I think this looks more clean compared to a `let` solution

Reply:
I don't like that. My problem with this solution is that you would naturally assume that the returned path can be used where ever the runtime path is required (e.g. when extracting the runtime archive). However, that's not the case. You can only use it in the context of LD path.

Reply:
We could also define `fetchRuntimePaths` as an internal method (e.g. inside the method where it is actually used):

```javascript
const fetchRuntime = () => {
   if (this._useRuntime) {
      return {
         runtimePath: `{this._wineEnginesDirectory}/runtime/lib/`,
         runtimePath64: `{this._wineEnginesDirectory}/runtime/lib64/`
      };
   } else {
      return { runtimePath: "", runtimePath64: "" };
   }
};
```

Would this solve the problem in your opinion?

Reply:
Yes it would but I still believe that we should not do it:
1. I don't really see the difference to the current implementation.
2. In general we should avoid using JS possibilities which might not be easily understandable for people who do not know JS (unless it gives us a big advantage).

Reply:
> I don't really see the difference to the current implementation.

The main difference is that it removes `let`

> In general we should avoid using JS possibilities which might not be easily understandable for people who do not know JS (unless it gives us a big advantage).

What exactly do you mean by JS possibilities? The "inner function" or the destructuring? Both are features that are also supported by a lot of other programming languages. Inner functions for example are also supported by Java:

```java
Runnable fetchRuntime = () => {
   ...
};
```

Reply:
I meant the inner function. We have to keep in mind that the people who get in contact with the scripts might know nothing about programming. The simpler we keep our scripts, the easier they will be understandable for them.

Reply:
I think certain parts of the scripts may be public but are normally used as blackbox components by the script users/developers. I count the engine implementations in this category. These classes/modules should be documented extensively so that the typical user won't have questions about what a method does.

Reply:
That's a valid point.